libs/queue/README.md

```diff
@@ -0,0 +1,64 @@
+# Overview
+
+The Queue library provides an implementation of a queue data structure. A Queue is a linear structure which follows the First-In-First-Out (FIFO) principle. This means that the elements added first are the ones that get removed first.
+
+The `Queue<T>` is a structure with a generic type parameter T. The elements in the queue are stored as a `Vec<T>`. The data structure is created empty and is growable.
+
+For more information please see the [specification](./SPECIFICATION.md).
+
+> **Note** The Queue<T> type in this library is different from built-in arrays and vectors due to its unique characteristics, especially its operation in FIFO manner.
+
+## Known Issues
+
+Currently, the Queue<T> type does not provide any way to iterate through its elements. 
+
+# Using the Library
+
+## Getting Started
+
+To use the Queue library, you need to add it to the Forc.toml file and then import it into your Sway project. For more details on how to add Sway-libs as a dependency to your Forc.toml, refer to the [README.md](../../../README.md).
```

Comment:
This is a broken README link. I recommend checking out where the file has been moved to and updating it - probably submit a PR to update other libraries if this file has been copied and pasted from another library